tests/manage/z_cluster/nodes/test_node_replacement_proactive.py

```diff
@@ -0,0 +1,102 @@
+
+        """
+
+        # Get worker nodes
+        worker_node_list = get_worker_nodes()
+        log.info(f"Current available worker nodes are {worker_node_list}")
+
+        osd_pods_obj = pod.get_osd_pods()
+        osd_node_name = pod.get_pod_node(random.choice(osd_pods_obj)).name
+        log.info(f"Selected OSD is {osd_node_name}")
+
+        log.info("Creating dc pod backed with rbd pvc and running io in bg")
+        for worker_node in worker_node_list:
+            if worker_node != osd_node_name:
+                rbd_dc_pod = dc_pod_factory(interface=constants.CEPHBLOCKPOOL, node_name=worker_node, size=20)
+                pod.run_io_in_bg(rbd_dc_pod, expect_to_fail=False, fedora_dc=True)
+
+        log.info("Creating dc pod backed with cephfs pvc and running io in bg")
+        for worker_node in worker_node_list:
+            if worker_node != osd_node_name:
+                cephfs_dc_pod = dc_pod_factory(interface=constants.CEPHFILESYSTEM, node_name=worker_node, size=20)
+                pod.run_io_in_bg(cephfs_dc_pod, expect_to_fail=False, fedora_dc=True)
+
+        # Unscheduling node
+        node.unschedule_nodes([osd_node_name])
+        log.info("Getting machine name from specified node name")
+        machine_name = machine.get_machine_from_node_name(osd_node_name)
+        log.info(f"Node {osd_node_name} associated machine is {machine_name}")
+        log.info(f"Deleting machine {machine_name}")
+        machine.delete_machine_and_check_state_of_new_spinned_machine(machine_name)
+        new_machine_list = machine.get_machines()
+        for machines in new_machine_list:
+            # Trimming is done to get just machine name
+            # eg:- machine_name:- prsurve-40-ocs-43-kbrvf-worker-us-east-2b-nlgkr
+            # After trimming:- prsurve-40-ocs-43-kbrvf-worker-us-east-2b
+            if re.match(machines.name[:-6], machine_name):
+                new_machine_name = machines.name
+        machineset_name = machine.get_machineset_from_machine_name(new_machine_name)
+        log.info("Wating for new worker node to be in ready state")
+        machine.wait_for_new_node_to_be_ready(machineset_name)
+        new_node_obj = node.get_node_from_machine_name(new_machine_name)
+        log.info("Adding ocs label to newly created worker node")
+        node_obj = ocp.OCP(kind='node')
+        node_obj.add_label(
+            resource_name=new_node_obj.name,
+            label=constants.OPERATOR_NODE_LABEL
+        )
+        log.info(
+            f"Successfully labeled {new_node_obj.name} with OCS storage label"
+        )
+        # Creating Resources
+        log.info("Creating Resources using sanity helpers")
+        self.sanity_helpers.create_resources(pvc_factory, pod_factory)
+        # Deleting Resources
+        self.sanity_helpers.delete_resources()
+        # Verify everything running fine
+        log.info("Verifying All resources are Running and matches expected result")
```

Comment:
We will also have to verify below,
*. Create DC app pods on the failure node and start IO before inducing failure.
*. Post failure, the DC app pods should automatically get created on the other nodes 
*. We should be able to start IO on the DC app pods.
FYI, I have a patch for doing above which is still under review, https://github.com/red-hat-storage/ocs-ci/pull/1591
I am okay if you have plans to cover this in another PR :)

Reply:
will cover in next pr when your pr gets merged